bundle/edu.gemini.pot/src/main/java/edu/gemini/spModel/gemini/gmos/GmosOiwfsGuideProbe.java

```diff
@@ -130,8 +129,9 @@ public PatrolField getPatrolField() {
     @Override
     public double calculateVignetting(final ObsContext ctx,
                                       final edu.gemini.spModel.core.Coordinates guideStarCoordinates) {
-        final ProbeArmGeometry probeArmGeometry       = GmosOiwfsProbeArm.instance();
-        return ScalaGuideProbeUtil.calculateVignetting(ctx, guideStarCoordinates, probeArmGeometry);
+        if (Site.GN.equals(ctx.getSite().getOrNull()))
+            ScalaGuideProbeUtil.calculateVignetting(ctx, guideStarCoordinates, GmosNorthOiwfsProbeArm$.MODULE$);
+        return ScalaGuideProbeUtil.calculateVignetting(ctx, guideStarCoordinates, GmosSouthOiwfsProbeArm$.MODULE$);
```

Comment:
`Site` is just an enum so we can get away with a reference (i.e, `==`) comparison.  A bigger issue is that the `if` branch calculation for `GN` appears to just be thrown away (you're too used to Scala I think).  Can the `Site` comparison just pick a guide probe arm and then you call the `calculateVignetting` with the selected arm?